src/package.cc

```diff
@@ -62,4 +62,79 @@ Package::Package(std::string name, double fill_min, double fill_max,
   }
 }
 
+TransportUnit::Ptr TransportUnit::unrestricted_ = NULL;
+
+TransportUnit::Ptr TransportUnit::Create(std::string name, int fill_min, int fill_max, std::string strategy) {
+  if (fill_min < 0 || fill_max < 0) {
+    throw ValueError("fill_min and fill_max must be non-negative");
+  }
+  else if (fill_min > fill_max) {
+    throw ValueError("fill_min must be less than or equal to fill_max");
+  }
+  Ptr t(new TransportUnit(name, fill_min, fill_max, strategy));
+  return t;
+}
+
+// singleton pattern: 
+// if the static member is not yet set, create a new object
+// otherwise return the object that already exists
+TransportUnit::Ptr& TransportUnit::unrestricted() {
+  if (!unrestricted_) {
+    unrestricted_ = Ptr(new TransportUnit(unrestricted_name_));
+  }
+  return unrestricted_;
+}
+
+int TransportUnit::GetTransportUnitFill(int qty) {
+  if (qty < fill_min_) {
+    // less than one pkg of material available
+    return 0;
+  }
+
+  if (strategy_ == "first") {
+    return fill_max_;
+  } else if (strategy_ == "equal") {
```

Comment:
This is a very restrictive strategy - maybe we need a (post-dissertation) dissertation strategy that is more like "balanced" where it ships as many as possible with as little variation between the fill of each unit as possible.

For example, you can't never ship an odd number of packages in an even number of containers with the "equal" strategy.